src/test/peerfinder/PeerFinder_test.cpp

```diff
@@ -157,6 +158,108 @@ class PeerFinder_test : public beast::unit_test::suite
         BEAST_EXPECT(n <= (seconds + 59) / 60);
     }
 
+    void
+    test_duplicateOutIn()
+    {
+        auto const seconds = 10000;
+        testcase("duplicate out/in");
+        TestStore store;
+        TestChecker checker;
+        TestStopwatch clock;
+        Logic<TestChecker> logic(clock, store, checker, journal_);
+        logic.addFixedPeer(
+            "test", beast::IP::Endpoint::from_string("65.0.0.1:5"));
+        {
+            Config c;
+            c.autoConnect = false;
+            c.listeningPort = 1024;
+            c.ipLimit = 2;
+            logic.config(c);
+        }
+
+        PublicKey const pk(randomKeyPair(KeyType::secp256k1).first);
+        std::size_t n = 0;
+
+        for (std::size_t i = 0; i < seconds; ++i)
```

Comment:
What scenarios are tested by iterating 10,000 times? Slots are added by `new_*_slot()` functions and removed by `remove()` in `on_closed()`. Those functions are only executed if the list is not empty. Basically the test is just repeated a few times.

Reply:
I do  not know. I just reuse the same logic as other tests in this suite.

Reply:
Not sure about other tests but it doesn't look like there is any benefit in repeating this test N times in this test. I think we just want to test `new_inbound_slot()`, `new_outbound_slot()`, and `on_closed()`.

Reply:
Up to you if you want to change it. I'm going to approve.